src/App.tsx

```diff
@@ -116,75 +126,97 @@ const hexRegex = /^#[a-f0-9]{6}$/gi;
 
 // TODO: add an error boundary for parse fails
 const Editor: Component<{ currentFile: FileSystemFileHandle }> = (props) => {
-  const [contents, { mutate }] = createResource(async () => {
+  const [obj, setObject] = createStore<Record<string, unknown>>({});
+  let lastUpdate = Date.now();
+  createEffect(async () => {
     const file = await props.currentFile.getFile();
     const text = await file.text();
-    console.log(parseJsonFile(text));
-    return text;
+    const json = JSON.parse(text);
+    setObject(reconcile(json));
+  });
+
+  createEffect(() => {
+    const interval = setInterval(async () => {
+      const file = await props.currentFile.getFile();
+      const text = await file.text();
+      const json = JSON.parse(text);
+
+      if (lastUpdate < file.lastModified) {
+        setObject(reconcile(json));
+        lastUpdate = Date.now();
+      }
+    }, 1000);
+    return () => clearInterval(interval);
   });
 
   const write = async (value: string) => {
-    mutate(value);
     // TODO: do we need to createWritable() and close() on every input or can
     // those be executed on file open / on cleanup instead?
     const file = await props.currentFile.createWritable();
     await file.write(value);
     await file.close();
-  }
+  };
 
-  const updateProperty = async (jsonString: string, key: string, value: string | number) => {
-    const newJson = JSON.parse(jsonString);
-    newJson[key] = value;
-    const newContents = JSON.stringify(newJson, null, 2);
+  const updateProperty = async (key: string, value: string | number) => {
+    setObject(reconcile({ ...obj, [key]: value }));
+    lastUpdate = Date.now();
```

Comment:
this is my first time seeing this pattern in solidjs code. I know if it were React, we'd need `lastUpdate` to be inside of a `useState()`... but I'm not sure if that's the case with solidjs. would this be overkill (or would there be any benefit to it)?

```ts
const [lastUpdate, setLastUpdate] = createSignal(Date.now());
```

Reply:
yeahh, i was debating between signal and raw variable for this. honestly, signal might be better in case we ever wanted to render `lastUpdate` to the UI somewhere in the future.

the best solution might be to combine this into our `obj` store.

```ts
type ObjStoreVal = {
  type: "loaded",
  value: Record<string, unknown>;
  lastUpdated: number
} | {
  type: "loading"
}
```

then we can correctly handle loading states as well.